src/AbstractAlgebra.jl

```diff
@@ -1229,13 +1233,25 @@ export Generic
 
 ###############################################################################
 #
-#   Polynomial Ring S, x = R["x"] syntax
+#   misc
+#
+###############################################################################
+
+include("misc/ProductIterator.jl")
+include("misc/Evaluate.jl")
+include("misc/VarNames.jl")
+
+###############################################################################
+#
+#   Polynomial Ring S, x = R[:x] syntax
 #
 ###############################################################################
 
 getindex(R::NCRing, s::VarName) = polynomial_ring(R, s)
-getindex(R::NCRing, s::VarName, ss::VarName...) =
-   polynomial_ring(R, [Symbol(x) for x in (s, ss...)])
+# `R[:x, :y]` returns `S, [x, y]` instead of `S, x, y`
+getindex(R::Ring, s1::VarNames, s::VarNames...) = ((S, x...) = polynomial_ring(R, s1, s...); (S, collect(x)))
+# Resolve ambiguity
+getindex(R::Ring, s::VarName) = polynomial_ring(R, s)
```

Comment:
```suggestion
getindex(R::NCRing, s1::VarNames, s::VarNames...) = ((S, x...) = polynomial_ring(R, s1, s...); (S, collect(x)))
```
Any reason why we don't just do this? Or, for that matter: why don't we just keep using the old implementation here, which just returns the "right" values from the start?
```suggestion
getindex(R::NCRing, s::VarName, ss::VarName...) =
   polynomial_ring(R, [Symbol(x) for x in (s, ss...)])
```
I am confused as to what this change here is gaining us?

Reply:
I thought it strange to read code calling `polynomial_ring(::NCRing, ::Vector{Symbol})` knowing it only exists for `Ring`. Though, this seems fine after all.

Using the generic `polynomial_ring`  interface allows to write `R[:x => 1:n]` resembling $R[x_1, \dots, x_n]$. On the other hand, this could confuse more than it helps. This is not Perl after all 😃

Reply:
But now the tests fail...